.github/workflows/ci.yml

```diff
@@ -0,0 +1,117 @@
+name: ci
+
+on:
+  # Makes it possible to trigger via the GitHub UI (Actions tab)
+  workflow_dispatch:
+  merge_group:
+  pull_request:
+  push:
+    branches:
+      - master
+
+defaults:
+  run:
+    shell: bash
+
+env:
+  RUST_BACKTRACE: 1
+  CARGO_INCREMENTAL: 0
+  CARGO_TERM_COLOR: always
+  RUSTDOCFLAGS: --deny warnings
+  RUSTFLAGS: --deny warnings
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref }}
+  cancel-in-progress: true
+
+jobs:
+  rust-test:
+    runs-on: ${{ matrix.os }}-latest
+
+    strategy:
+      matrix:
+        os: [ubuntu, windows, macos]
+
+    steps:
+      - uses: actions/checkout@v3
+      - uses: actions-rust-lang/setup-rust-toolchain@v1
+
+      - run: cargo test --locked
+
+  rust-lint:
+    runs-on: ${{ matrix.os }}-latest
+
+    strategy:
+      matrix:
+        os: [ubuntu, windows, macos]
+
+    steps:
+      - uses: actions/checkout@v3
+      - uses: actions-rust-lang/setup-rust-toolchain@v1
+
+      - run: cargo clippy
+      - run: cargo doc --no-deps
+
+  rust-formatting:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      # We don't need any cache here, so use dtolnay action directly
+      - uses: dtolnay/rust-toolchain@stable
```

Comment:
Do we maybe want to pin the stable version as well, to ensure that they build on stable v1.64 or something like that? I think this would at least make sense for `cargo-marker`. Not a blocker for this PR, just a question :)

Reply:
I also thought that it makes sense to pin the toolchain version. I think it makes sense to select the latest stable version target for `cargo-marker` once we add pre-compiled binaries to the release process.